test_suite.py

```diff
@@ -1,126 +1,213 @@
-from rtps_test_utilities import ReturnCode
-
+from rtps_test_utilities import ReturnCode, log_message
+import re
+import pexpect
 # rtps_test_suite_1 is a dictionary where we define the TestSuite
 # (with its TestCases that we will test in interoperability_report.py).
 # The dictionary has the following structure:
-#       'name' : [parameters_publisher, parameters_subscriber,
-#              expected_return_code_publisher, expected_return_code_subscriber]
+#       'name' : [[parameters], [expected_return_codes], <OPTIONAL>:function]
 # where:
 #       * name: TestCase's name (defined by us)
-#       * parameters_publisher: parameters the shape_main application
-#         uses for the publisher process
-#       * parameters_subscriber: parameters the shape_main application
-#         uses for the subscriber process
-#       * expected_return_code_publisher: expected publisher ReturnCode
-#         for a succeed test execution.
-#       * expected_return_code_subscriber: expected subscriber ReturnCode
+#       * parameters: list with parameters for the shape_main application.
+#       * expected_return_codes: list with expected ReturnCodes
 #         for a succeed test execution.
+#       * function[OPTIONAL]: function to check how the Subscribers receive
+#         the samples from the Publishers. By default it does not check
+#         anything. It has to be implemented by us.
+# The number of elements in parameters will define how many shape_main
+# application we will run. It should be the same as the number of
+# elements in expected_return_codes.
 #
-# There are also two testCases that contains more parameters:
-#       Test_Ownership_3 and Test_Ownership_4.
-# That is because these two cases run two publishers and one subscriber.
-# These two cases are handled manually in the interoperability_report.py
-# script.
-# The parameters in this case are:
-#       * name: TestCase's name (defined by us)
-#       * parameters_publisher1: parameters the shape_main application
-#         uses for the publisher1 process
-#       * parameters_publisher2: parameters the shape_main application
-#         uses for the publisher2 process
-#       * parameters_subscriber: parameters the shape_main application
-#         uses for the subscriber process
-#       * expected_return_code_publisher1: expected publisher1 ReturnCode
-#         for a succeed test execution.
-#       * expected_return_code_publisher2: expected publisher2 ReturnCode
-#         for a succeed test execution.
-#       * expected_return_code_subscriber: expected subscriber ReturnCode
-#         for a succeed test execution.
+# test_ownership3_4 and test_reliability_4 are two functions defined
```

Comment:
let's review the documentation in a meeting

Reply:
okey